lib/ui/compositing/scene_builder.cc

```diff
@@ -276,10 +277,17 @@ void SceneBuilder::addPicture(double dx,
                               double dy,
                               Picture* picture,
                               int hints) {
-  auto layer = std::make_unique<flutter::PictureLayer>(
-      SkPoint::Make(dx, dy), UIDartState::CreateGPUObject(picture->picture()),
-      !!(hints & 1), !!(hints & 2));
-  AddLayer(std::move(layer));
+  if (picture->picture()) {
+    auto layer = std::make_unique<flutter::PictureLayer>(
+        SkPoint::Make(dx, dy), UIDartState::CreateGPUObject(picture->picture()),
+        !!(hints & 1), !!(hints & 2));
+    AddLayer(std::move(layer));
+  } else {
+    auto layer = std::make_unique<flutter::DisplayListLayer>(
+        SkPoint::Make(dx, dy), picture->display_list(), !!(hints & 1),
+        !!(hints & 2));
+    AddLayer(std::move(layer));
+  }
```

Comment:
Can we just make this a method on `Picture` instead?

e.g. `auto layer = picture->createLayer();`?

And elsewhere: can we generally avoid needing multiple callsites to be aware of whether they need to ask for the `picture()` or the `display_list()`.

Reply:
I only remember 2 places that dealt with this. Here and the unittest that used to assume that all pictures were SkP. The latter shouldn't leak methods into the Picture as it is just a unittest.

Reply:
Another issue is that Picture is in lib/ui/painting so having a method to produce a flow/layer/*Layer on it would be out of place. AFAIR, code in lib/ui isn't really aware of layers (but there is some cross-talk to flow/*?).

Reply:
When SkPicture goes away here, we won't need an if or a method on Picture to manage this distinction. Since it is only 3 lines, I think I'm going to leave in the 2 pieces of code to reduce the cross-file technical debt here for the eventual elimination of SkPicture.